### DISCUSSION TITLE
You want platform agnosticism rather than sourcehut.

### DISCUSSION THREAD

**Discussion by amano-kenji** [hannesbraun/airwindows-lv2#13] [General]
It's good to not rely on big tech platforms like github. But, I would not consider sourcehut a sanctuary of freedom.

Drew DeVault announced he is going to remove unapproved cryptocurrency projects from sourcehut. This is censorship. He thinks cryptocurrency is only used to scam people. Well, fiat currencies are used very often to scam people and even kill people. My relatives gave a lot of fiat currency to scam artists.

He thinks tools harm people when, in reality, it is people who use tools to harm people. We need to educate people instead of banning tools. He's misguided. Banning cryptocurrency doesn't make anyone safe. Rather, wars and various crimes have been funded by US dollars. Cryptocurrency hasn't funded any war. Wars could have been downsized a lot if there was no US dollar. Should we then ban US dollars to save millions of people? No! It is people who use US dollars with evil mindset who are the problems. Evil people will find a way to use any available tool to harm people. Evil people are evil because they have received bad education. Most evil people don't know they are evil because they were taught badly. They think it's okay to harm others in return for money. That's why the world is hell today. Some evil people know they are evil, but their brains are in degraded states.

GitHub has its own shares of censorship.  Codeberg.org claims to be independent, but it also succumbed to law enforcement and censored various projects including wikiless, a wikipedia frontend.

Platforms that claim to be free engage in censorship because they are under duress, or the management voluntarily sacrifices freedom in the name of safety while, in reality, they don't really make anyone safe through censorship.

If you really want freedom, then you should aim for platform agnosticism. For example, your mailing list is hosted on sourcehut. While your project is not likely to be censored by sourcehut, if you want platform agnosticism, you don't want to be dependent on any one platform run by others for issues and patches.

I'm not sure that sourcehut is less likely to censor projects because it censored projects even though it was not under duress. At least, codeberg censored projects under duress by law enforcement.

Perhaps, you may run your own gitea instance, or your own mailing list on your rented servers.

Do not believe in people who claim to value freedom when their actions tell you they don't actually value freedom. Anyone can make claims without meaning them. In the current condition, don't see any platform as a sanctuary of freedom. Only education will make people moral and righteous in the long run.

**Comment by hannesbraun** [comment]
Thanks for taking the time to write this.

> Perhaps, you may run your own gitea instance, or your own mailing list on your rented servers.

I initially thought about doing this. If I would host something myself, I'd really prefer sourcehut over Gitea/Forgejo (or Gitlab) due to various reasons. However, running my own public git server is exceeding the time I currently have. Maybe that is something for the future.

That's why I chose sourcehut. In my opinion it's certainly better than GitHub. And while harmful crypto-related projects are banned from sourcehut, I can still totally live with this. Other than this, I'm more than happy with choosing sourcehut. And my opinions are often really overlapping with the ones from Drew. Is banning crypto projects the best option? I don't know. But I can see a good will in their actions.

So for now, I'll keep on using sourcehut.

**Reply by amano-kenji** [reply]
Censoring any tool is always bad. The first consideration is that people have a right to use any tool as long as they are not initiating harm on others with the tool. There is no such thing as harmful crypto projects. It is just a neutral tool that can be used for good or evil. There are only people who initiate harm on others with crypto projects. The fact that there are people who scam others with cryptocurrencies doesn't mean the crypto projects are inherently bad. It's badly educated people who are bad. Would you say a fiat currency is harmful because it is used a lot by scammers or war criminals? If a tyrant creates a fiat currency to control or scam people, is it a harmful fiat currency? If a person makes a kitchen knife in order to kill an innocent person, is it a harmful knife? Or, is the person a harmful person? If a person learns kickboxing to bully someone else, is kickboxing a harmful martial art?

I think he was misinformed by mainstream media which always side with banks, and banks hate cryptocurrencies because cryptocurrencies are not under direct control of banks. Banks want to be the only source of money. They don't want competitions. I think he was misled into helping banks by listening only to mainstream media outlets.

While you are probably not going to be affected on any platform, if you care about freedom, in the future, you may consider moving your mailing list to another server run by someone else who at least doesn't engage in voluntary censorship. If all you use is a remote git viewer and a mailing list, then you don't technically need sourcehut. You can use other softwares. Linux kernel doesn't require sourcehut for its operation.
